src/instruments/yokogawa/yokogawa6370.py

```diff
@@ -60,19 +57,36 @@ def __init__(self, parent, idx):
 
         # METHODS #
 
-        def data(self, bin_format=True):
-            cmd = f":TRAC:Y? {self._name}"
+        def _data(self, axis, limits=None, bin_format=True):
+            """Get data of `axis`."""
+            if limits is None:
+                cmd = f":TRAC:{axis}? {self._name}"
+            elif isinstance(limits, (tuple, list)) and len(limits) >= 2:
+                cmd = f":TRAC:{axis}? {self._name},{limits[0]+1},{limits[1]+1}"
+            else:
+                raise AssertionError(
```

Comment:
I don't think this should raise an `AssertionError`, since no `assert` statement is actually here. 
If the `isinstance()` fails, a `TypeError` should be the result. If the length is wrong, the most appropriate would probably be an `IndexError`, `ValueError` could be a second idea for that case.

Reply:
I changed it to `ValueError`, as the given value for `limits` is invalid.